spring/src/main/java/org/axonframework/spring/config/SpringAxonAutoConfigurer.java

```diff
@@ -244,8 +245,11 @@ private void registerCorrelationDataProviders(Configurer configurer) {
     }
 
     private void registerEventUpcasters(Configurer configurer) {
-        Arrays.stream(beanFactory.getBeanNamesForType(EventUpcaster.class))
-              .forEach(name -> configurer.registerEventUpcaster(c -> getBean(name, c)));
+        beanFactory.getBeansOfType(EventUpcaster.class)
```

Comment:
We need to be very cautious using `getBeans` operations, as they cause (eager) initialization of the bean instances.
Instead, since we're using a ConfigurableListableBeanFactory, we can use `getType(String name)` to get the actual type of the bean instance, and use the comparator on the class.  Looking at the implementation of `AnnotationAwareOrderComparator`, it already takes into account that it may receive a Class (or any other AnnotatedElement, for that matter)

Reply:
Great spot! 

I'll adjust the format of this method to:
- Retrieve the bean names
- Based on names, get the types
- Sort the types using the `AnnotationAwareOrderComparator`
- Set the bean retrieval operations with the upcaster-builder provided to `Configurer#registerEventUpcaster`.